File: sample-java/src/main/java/com/android/mvvmarchitecturejavasample/TestViewModel.java
package com.android.mvvmarchitecturejavasample;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.LifecycleOwner;

import com.imyyq.mvvm.base.BaseModel;
import com.imyyq.mvvm.base.BaseViewModel;

import org.jetbrains.annotations.NotNull;

public class TestViewModel extends BaseViewModel<BaseModel> {
    public TestViewModel(@NotNull Application app) {
        super(app);
    }

    @Override
    public void onResume(@NonNull LifecycleOwner owner) {
    }
}

File: sample/src/main/java/com/imyyq/sample/TestViewModel.kt
package com.imyyq.sample

import android.app.Application
import com.imyyq.mvvm.base.BaseModel
import com.imyyq.mvvm.base.BaseViewModel
import com.imyyq.mvvm.utils.LogUtil

class TestViewModel(app: Application) : BaseViewModel<BaseModel>(app) {
    fun test() {
        LogUtil.i("TestViewModel", "commonLog - test: ")
    }
}